### ISSUE TITLE
TLS Handshake Error

### ISSUE THREAD

**Issue by alastairs** [containous/traefik#1827]
### Do you want to request a *feature* or report a *bug*?

Report a bug.

### What did you do?

We are running traefik in Docker on Kubernetes, and K8s is deployed on Azure Container Service. We see the following error appearing in the console output approximately once a minute:

```
2017-07-05T12:38:59.832673311Z 2017/07/05 12:38:59 server.go:2753: http: TLS handshake error from 10.240.0.6:58147: EOF 
```

Having turned on debug logging, I see it is always accompanied by this log message:
```
2017-07-05T12:38:59.718337351Z time="2017-07-05T12:38:59Z" level=debug msg="Skipping event from kubernetes &Endpoints{ObjectMeta:ObjectMeta{Name:kube-scheduler,GenerateName:,Namespace:kube-system,SelfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler,UID:673d8e35-5c66-11e7-b220-000d3a216355,ResourceVersion:10786823,Generation:0,CreationTimestamp:2017-06-29 01:00:54 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{control-plane.alpha.kubernetes.io/leader: {"holderIdentity":"k8s-master-16396601-0","leaseDurationSeconds":15,"acquireTime":"2017-06-29T01:00:54Z","renewTime":"2017-07-05T12:38:59Z","leaderTransitions":0},},OwnerReferences:[],Finalizers:[],ClusterName:,},Subsets:[],}"  
```

As per [this thread](https://traefik.slack.com/archives/C0CDT22PJ/p1499179044157305), this appears to be a new instance of #1025. Although the line number has changed, it remains well outside the range of possible lines in `server.go`. Additionally the IP addresses lie outside of the cluster subnet. 

### What did you expect to see?

traefik ticking along nicely, some informational messages, no TLS handshake messages.

### What did you see instead?
These error messages. They're particularly annoying as they don't appear to be log messages, and so can't be turned down, turned off, or filtered out.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.3.2
Codename:     raclette
Go version:   go1.8.3
Built:        2017-06-29_04:52:35PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

We are running traefik:1.3 from Docker on Kubernetes 1.5.3, running on Azure Container Service.

`docker images traefik:1.3` output:
```
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
traefik             1.3                 917039496236        5 days ago          45MB
```

`kubectl version` output:json
```
Server Version: version.Info{Major:"1", Minor:"5", GitVersion:"v1.5.3", GitCommit:"029c3a408176b55c30846f0faedf56aae5992e9b", GitTreeState:"clean", BuildDate:"2017-02-15T06:34:56Z", GoVersion:"go1.7.4", Compiler:"gc", Platform:"linux/amd64"}
```

Kubernetes Deployment for traefik is (including traefik args at the bottom):
```yaml
kind: Deployment
apiVersion: extensions/v1beta1
metadata:
  name: traefik-controller
  labels:
    service: traefik
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: traefik-ingress-lb
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb
        name: traefik-ingress-lb
        service: traefik
    spec:
      terminationGracePeriodSeconds: 60
      volumes:
        - name: config
          configMap:
            name: traefik-conf
        - name: acme
          hostPath:
            path: /acme
      containers:
      - image: traefik:1.3
        name: traefik-ingress-lb
        resources:
          limits:
            cpu: 200m
            memory: 30Mi
          requests:
            cpu: 100m
            memory: 20Mi
        volumeMounts:
        - mountPath: "/config"
          name: "config"
        - mountPath: "/acme"
          name: "acme"
        ports:
        - containerPort: 80
          hostPort: 80
        - containerPort: 443
          hostPort: 443
        - containerPort: 8080
          hostPort: 8080
        args:
        - --configfile=/config/traefik.toml
        - --web
        - --kubernetes
        - --kubernetes.namespaces={{K8S_NAMESPACE}}
        - --loglevel=INFO
```

```toml
defaultEntryPoints = ["http","https"]
[file]
watch=true
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]
[web]
address = ":8080"
ReadOnly = true
  [web.statistics]
  RecentErrors = 10
[acme]
email = "{{REDACTED}}"
storageFile = "/acme/acme-ssl.json"
entryPoint = "https"
# caServer = "https://acme-staging.api.letsencrypt.org/directory"
OnHostRule = true
```

### If applicable, please paste the log output in debug mode (`--debug` switch)
I can provide a full log file privately; please DM [@alastairs on Slack](https://traefik.slack.com/team/alastairs) if you would like it.

**Comment by timoreimann** [comment]
The log line is apparently from [`net/http`'s server.go](https://golang.org/src/net/http/server.go?s=69188:72198#L2753). If it helps, someone could hack the file temporarily to add more logging / dump a stack trace in order to identify the root cause.

**Comment by michaelchiche** [comment]
I am currently having the same issue in a docker swarm cluster in Digital Ocean

**Comment by kcrawley** [comment]
Seeing the same issue using Docker EE 17.06.1-ee-2 running on Ubuntu 16.04.03/vmware, I've also seen this on Swarm CE 17.06 running on Ubuntu 16.04/vmware. It doesn't appear to cause any critical issues, just log noise.

**Comment by emilevauge** [comment]
@alastairs @michaelchiche Do you see any issue related to this log ? As @kcrawley said, I think it's just some log noise.

**Comment by alastairs** [comment]
> Do you see any issue related to this log 
Yeah, we ship our logs to Elastic Search, and the large amount of noise puts a ton of extra stress on our cluster that we'd rather avoid.

**Comment by MichaelErmer** [comment]
Also seeing this with `v1.4.0 / roquefort`, only the line number changed to 2848.

We are seeing several hundred of these messages per second at (~10% of the current throughput), the ip is the one of the one of the user requesting data.

**Comment by johnhamelink** [comment]
I'm also seeing this issue on Kubernetes 1.7 alongside a "skipping event" log item during the ACME certification retrieval process.

```
time="2017-11-06T15:44:30Z" level=error msg="Service not found for default/dandy-duck-<redacted: chart name was here>"
time="2017-11-06T15:44:30Z" level=debug msg="Skipping event from kubernetes &Endpoints{ObjectMeta:ObjectMeta{Name:kube-scheduler,GenerateName:,Namespace:kube-system,SelfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler,UID:18fd1002-bef5-11e7-8e7d-0ae8f2922d04,ResourceVersion:589500,Generation:0,CreationTimestamp:2017-11-01 11:09:15 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{control-plane.alpha.kubernetes.io/leader: {"holderIdentity":"ip-10-0-45-122","leaseDurationSeconds":15,"acquireTime":"2017-11-03T15:31:04Z","renewTime":"2017-11-06T15:44:30Z","leaderTransitions":1},},OwnerReferences:[],Finalizers:[],ClusterName:,},Subsets:[],}"
time="2017-11-06T15:44:30Z" level=debug msg="Look for provided certificate to validate [api-staging.<redacted>.com]..."
time="2017-11-06T15:44:30Z" level=debug msg="No provided certificate found for domains [api-staging.<redacted>.com], get ACME certificate."
time="2017-11-06T15:44:30Z" level=debug msg="Challenge GetCertificate api-staging.<redacted>.com"
time="2017-11-06T15:44:30Z" level=debug msg="ACME got domain cert api-staging.<redacted>.com"
2017/11/06 15:44:30 server.go:2848: http: TLS handshake error from 10.0.72.159:48876: EOF
time="2017-11-06T15:44:30Z" level=debug msg="Look for provided certificate to validate [api-staging.<redacted>.com]..."
time="2017-11-06T15:44:30Z" level=debug msg="No provided certificate found for domains [api-staging.<redacted>.com], get ACME certificate."
time="2017-11-06T15:44:30Z" level=debug msg="Challenge GetCertificate api-staging.<redacted>.com"
time="2017-11-06T15:44:30Z" level=debug msg="ACME got domain cert api-staging.<redacted>.com"
2017/11/06 15:44:31 server.go:2848: http: TLS handshake error from 10.0.72.159:48878: EOF
time="2017-11-06T15:44:32Z" level=debug msg="Received event from kubernetes &Endpoints{ObjectMeta:ObjectMeta{Name:kube-scheduler,GenerateName:,Namespace:kube-system,SelfLink:/api/v1/namespaces/kube-system/endpoints/kube-scheduler,UID:18fd1002-bef5-11e7-8e7d-0ae8f2922d04,ResourceVersion:589501,Generation:0,CreationTimestamp:2017-11-01 11:09:15 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{control-plane.alpha.kubernetes.io/leader: {"holderIdentity":"ip-10-0-45-122","leaseDurationSeconds":15,"acquireTime":"2017-11-03T15:31:04Z","renewTime":"2017-11-06T15:44:32Z","leaderTransitions":1},},OwnerReferences:[],Finalizers:[],ClusterName:,},Subsets:[],}"
```

Traefik does actually work, except SSL certificates are not used properly (I get a NET::ERR_CERT_AUTHORITY_INVALID error in chrome). Firefox gives the following info:

```
https://<redacted>.com/docs

Peer’s Certificate issuer is not recognized.

HTTP Strict Transport Security: false
HTTP Public Key Pinning: false

Certificate chain:

-----BEGIN CERTIFICATE-----
MIIF+jCCBOKgAwIBAgITAPrxPPwM/VAWTNiY/miE+ceI7zANBgkqhkiG9w0BAQsF
ADAiMSAwHgYDVQQDDBdGYWtlIExFIEludGVybWVkaWF0ZSBYMTAeFw0xNzExMDYx
MzQ3NTBaFw0xODAyMDQxMzQ3NTBaMCYxJDAiBgNVBAMTG2FwaS1zdGFnaW5nLmZp
bmRhcGxheWVyLmNvbTCCAiIwDQYJKoZIhvcNAQEBBQADggIPADCCAgoCggIBAMVQ
nAYDeBqf5C0fA0vvMSKWMUFZfxTX9U4XnrfgSLy7SV93oknzeck5NoNkwaKf2/LV
bM830FFWaDl7tn5DyQooINSyusRHhmLXnhjGkuHg/esXzml9wmOkAYAEJ+tZ80G0
IzSPY3OLTg5xVLquUEh4VOoLVRXoqPDV527/ZsUTGWb8lQV3YYOnghFQdVfpD2UD
h4vxlfvBnjmkaURmYksRQxf2mMS8o4xx0weHwp3vBynlctiqLbGAryizroGxLuzq
hH9pB59zdiMN8x7f4HvYab7KgQ/ywBKWYcfONYfsgpEcqE1G6bNVZ8MHVdsIEd/z
El0gU/EkTIqaO9Oq9SZtzUED97JpnCr/wJym2Jy1lvu7Js0TInbMuH5/T4iIulhv
tsu4k6oqeE3pwWxg7oPqLFW8O4SJI5ERVplsgC8r8j9zhzwLGYL3xWrPtA8hk631
IS5G4HfxqjAuMdK0NDxbBrk82VCWe4aPXasHiYtK3yiiKcaM65tL15E3DtBcJ5WF
yfz5fjp0t+/S+Ds2keIHSaWs5rnnEseTly9Af4GnO4hCxi/0I4KW2R6o+cMU3yP1
xTpMCz4mjGMJdbQeW/tU/2OgbjThqTlAvGTaDEx3zqxcYv21iT9c919Re5xyjjcm
QmIWq0j+zrkaPbvbyPUo18h7YRz2g3cIKzjmRnFVAgMBAAGjggIjMIICHzAOBgNV
HQ8BAf8EBAMCBaAwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMCMAwGA1Ud
EwEB/wQCMAAwHQYDVR0OBBYEFE9UyLwFLxsDWk+Uk/ATQa2NiVYcMB8GA1UdIwQY
MBaAFMDMA0a5WCDMXHJw8+EuyyCm9Wg6MHcGCCsGAQUFBwEBBGswaTAyBggrBgEF
BQcwAYYmaHR0cDovL29jc3Auc3RnLWludC14MS5sZXRzZW5jcnlwdC5vcmcwMwYI
KwYBBQUHMAKGJ2h0dHA6Ly9jZXJ0LnN0Zy1pbnQteDEubGV0c2VuY3J5cHQub3Jn
LzAmBgNVHREEHzAdghthcGktc3RhZ2luZy5maW5kYXBsYXllci5jb20wgf4GA1Ud
IASB9jCB8zAIBgZngQwBAgEwgeYGCysGAQQBgt8TAQEBMIHWMCYGCCsGAQUFBwIB
FhpodHRwOi8vY3BzLmxldHNlbmNyeXB0Lm9yZzCBqwYIKwYBBQUHAgIwgZ4MgZtU
aGlzIENlcnRpZmljYXRlIG1heSBvbmx5IGJlIHJlbGllZCB1cG9uIGJ5IFJlbHlp
bmcgUGFydGllcyBhbmQgb25seSBpbiBhY2NvcmRhbmNlIHdpdGggdGhlIENlcnRp
ZmljYXRlIFBvbGljeSBmb3VuZCBhdCBodHRwczovL2xldHNlbmNyeXB0Lm9yZy9y
ZXBvc2l0b3J5LzANBgkqhkiG9w0BAQsFAAOCAQEAX8XVCriVErAW09aaGkAw+ZsZ
nebQbCKe+2Fyb4tA+f5bVNizI4ptxg1rifJievfmThJHWvKMU7H4zVP56pC9fOll
yF4UiGXHrWbEucZJiYJWm6CVcUrVJY9q+X/eOUI173WKNIlSHyftJQrM0r80G9mS
4v6lGtWbguiSSkDACnjP4KSwTwaeqA8qZ7FwW5TduE8qFVzcaN5/8dvONQpWaD1t
/aH1xgQj1KzgzFed3bl0498UH0Cu6KjHm8XtyuEiRfpUPg4TxeFruLwt5qY8X2MF
6pwg56Hr5wGCR7fU7whqd4E7mH/8uLv5InGUI14shVSY3qdhl4VNW2B7J9Pz2Q==
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
MIIEqzCCApOgAwIBAgIRAIvhKg5ZRO08VGQx8JdhT+UwDQYJKoZIhvcNAQELBQAw
GjEYMBYGA1UEAwwPRmFrZSBMRSBSb290IFgxMB4XDTE2MDUyMzIyMDc1OVoXDTM2
MDUyMzIyMDc1OVowIjEgMB4GA1UEAwwXRmFrZSBMRSBJbnRlcm1lZGlhdGUgWDEw
ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQDtWKySDn7rWZc5ggjz3ZB0
8jO4xti3uzINfD5sQ7Lj7hzetUT+wQob+iXSZkhnvx+IvdbXF5/yt8aWPpUKnPym
oLxsYiI5gQBLxNDzIec0OIaflWqAr29m7J8+NNtApEN8nZFnf3bhehZW7AxmS1m0
ZnSsdHw0Fw+bgixPg2MQ9k9oefFeqa+7Kqdlz5bbrUYV2volxhDFtnI4Mh8BiWCN
xDH1Hizq+GKCcHsinDZWurCqder/afJBnQs+SBSL6MVApHt+d35zjBD92fO2Je56
dhMfzCgOKXeJ340WhW3TjD1zqLZXeaCyUNRnfOmWZV8nEhtHOFbUCU7r/KkjMZO9
AgMBAAGjgeMwgeAwDgYDVR0PAQH/BAQDAgGGMBIGA1UdEwEB/wQIMAYBAf8CAQAw
HQYDVR0OBBYEFMDMA0a5WCDMXHJw8+EuyyCm9Wg6MHoGCCsGAQUFBwEBBG4wbDA0
BggrBgEFBQcwAYYoaHR0cDovL29jc3Auc3RnLXJvb3QteDEubGV0c2VuY3J5cHQu
b3JnLzA0BggrBgEFBQcwAoYoaHR0cDovL2NlcnQuc3RnLXJvb3QteDEubGV0c2Vu
Y3J5cHQub3JnLzAfBgNVHSMEGDAWgBTBJnSkikSg5vogKNhcI5pFiBh54DANBgkq
hkiG9w0BAQsFAAOCAgEABYSu4Il+fI0MYU42OTmEj+1HqQ5DvyAeyCA6sGuZdwjF
UGeVOv3NnLyfofuUOjEbY5irFCDtnv+0ckukUZN9lz4Q2YjWGUpW4TTu3ieTsaC9
AFvCSgNHJyWSVtWvB5XDxsqawl1KzHzzwr132bF2rtGtazSqVqK9E07sGHMCf+zp
DQVDVVGtqZPHwX3KqUtefE621b8RI6VCl4oD30Olf8pjuzG4JKBFRFclzLRjo/h7
IkkfjZ8wDa7faOjVXx6n+eUQ29cIMCzr8/rNWHS9pYGGQKJiY2xmVC9h12H99Xyf
zWE9vb5zKP3MVG6neX1hSdo7PEAb9fqRhHkqVsqUvJlIRmvXvVKTwNCP3eCjRCCI
PTAvjV+4ni786iXwwFYNz8l3PmPLCyQXWGohnJ8iBm+5nk7O2ynaPVW0U2W+pt2w
SVuvdDM5zGv2f9ltNWUiYZHJ1mmO97jSY/6YfdOUH66iRtQtDkHBRdkNBsMbD+Em
2TgBldtHNSJBfB3pm9FblgOcJ0FSWcUDWJ7vO0+NTXlgrRofRT6pVywzxVo6dND0
WzYlTWeUVsO40xJqhgUQRER9YLOLxJ0O6C8i0xFxAMKOtSdodMB3RIwt7RFQ0uyt
n5Z5MqkYhlMI3J1tPRTp1nEt9fyGspBOO05gi148Qasp+3N+svqKomoQglNoAxU=
-----END CERTIFICATE-----
```

**Comment by ldez** [comment]
@johnhamelink The symptoms you are experiencing differ from those of the original bug submitter.

Please ask your question on:
- [Stack Overflow](https://stackoverflow.com/questions/tagged/traefik) using the "traefik" tag
- the [Traefik community Slack channel](https://traefik.herokuapp.com)

**Comment by johnhamelink** [comment]
@ldez Yes you are right, I have removed my previous comment as it would probably cause more confusion than anything else!

**Comment by MichaelErmer** [comment]
@ldez how can i assist with more info on this?

**Comment by ldez** [comment]
@MichaelErmer I have already analyze the problem. It's in Go itself.
For now I don't have a clean solution.

**Comment by traefiker** [comment]
Closed by #2420.
